src/ducks/context/SelectionContext.jsx

```diff
@@ -15,6 +15,31 @@ export const useSelectionContext = () => {
   return useContext(SelectionContext)
 }
 
+export const useSelectionToggler = item => {
+  const {
+    addToSelection,
+    isSelected,
+    removeFromSelection,
+    isSelectionModeEnabled
+  } = useSelectionContext()
+
+  const isItemSelected = useMemo(() => isSelected(item), [isSelected, item])
+
+  const toggleItemSelection = useCallback(() => {
+    if (isSelectionModeEnabled) {
+      isItemSelected ? removeFromSelection(item) : addToSelection(item)
+    }
+  }, [
+    isSelectionModeEnabled,
+    isItemSelected,
+    removeFromSelection,
+    item,
+    addToSelection
+  ])
```

Comment:
I wonder why there is so much dependencies, it might be a hint that the structure is not correct. 
A toggle should only care about its value, not about other stuff IMHO. It might be easier not to create another context but to add the toggleSelection to the selection context.

Also, I have the impression that when toggling a row, every other row will be re-rendered since removeSelection will be changed (since removeSelection has "selected" in dependencies).

Reply:
Oui je suis bien d'accord 👍 , j'avais pas trop réussi à simplifier du premier coup. J'ai déplacé le `toggleSelection` dans le `useSelectionContext` qui retourne un `{ toggleSelection, isItemSelected, ...selectionContext }`

En l'état toutes les lignes sont re-rendues quand "selected" change en effet... 🤔

Reply:
Oui c'est l'idée, après ca dépend du nombre de ligne rendues, je trouve ca juste dommage qu'on ne re-rende tout alors qu'il n'y a pas besoin. Après si c'est fluide en l'état, pourquoi pas...

Si la ligne ne recevait qu'un callback pour toggle et son état (et pas tout l'état de toute les lignes), elle n'aurait pas à se re-rendre.